Automattic/jetpack
# Instant Search: Add redux state for API, search, and sort

Replaces #18099.

#### Changes proposed in this Pull Request:
* Adds Redux actions, effects, reducers, and selectors for handling API requests, setting user-inputted search queries, and selected sort type.
* The Redux store is not connected to the component tree in this PR; this will occur in a future PR.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Review the code and ensure it looks reasonable.
* Run `yarn test-search` and ensure all tests pass.

#### Proposed changelog entry for your changes:
* None.
